test/jdk/javax/swing/JToolTip/bug5047379.java

```diff
@@ -0,0 +1,170 @@
+ *
+ * This code is distributed in the hope that it will be useful, but WITHOUT
+ * ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
+ * FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
+ * version 2 for more details (a copy is included in the LICENSE file that
+ * accompanied this code).
+ *
+ * You should have received a copy of the GNU General Public License version
+ * 2 along with this work; if not, write to the Free Software Foundation,
+ * Inc., 51 Franklin St, Fifth Floor, Boston, MA 02110-1301 USA.
+ *
+ * Please contact Oracle, 500 Oracle Parkway, Redwood Shores, CA 94065 USA
+ * or visit www.oracle.com if you need additional information or have any
+ * questions.
+ */
+/* @test
+   @key headful
+   @bug 5047379
+   @summary Checks that Tooltips are rendered properly for Metal Look and Feel
+   @library ../regtesthelpers
+   @build Util
+   @run main bug5047379
+*/
+
+import javax.swing.JButton;
+import javax.swing.JComponent;
+import javax.swing.JFrame;
+import javax.swing.JPanel;
+import javax.swing.JTextArea;
+import javax.swing.JToolTip;
+import javax.swing.KeyStroke;
+import javax.swing.plaf.metal.MetalToolTipUI;
+import javax.swing.SwingUtilities;
+import javax.swing.UIManager;
+
+import java.awt.BorderLayout;
+import java.awt.Dimension;
+import java.awt.event.KeyEvent;
+import java.awt.GridLayout;
+import java.awt.Point;
+import java.awt.Robot;
+
+public class bug5047379 {
+    static JFrame frame;
+    static JButton a;
+    static JButton b;
+    static JButton c;
+    static JButton d;
+    static Robot testRobot;
+
+    public static void main(String[] args) throws Exception {
+        UIManager.setLookAndFeel("javax.swing.plaf.metal.MetalLookAndFeel");
+        testRobot = new Robot();
+        SwingUtilities.invokeAndWait(new Runnable() {
+            public void run() {
+                runTest();
+            }
+        });
+        testRobot.delay(1000);
+        testRobot.waitForIdle();
+        Point movePoint = getButtonPoint(b);
+        testRobot.mouseMove(movePoint.x, movePoint.y);
+        testRobot.delay(2000);
+        testRobot.waitForIdle();
+        handleToolTip();
+    }
+
+
+    static void handleToolTip() throws Exception {
+        SwingUtilities.updateComponentTreeUI(frame);
+
+        SwingUtilities.invokeAndWait(new Runnable() {
+            @Override
+            public void run() {
+                try {
+                    JToolTip tooltip = (JToolTip) Util.findSubComponent(
+                            JFrame.getFrames()[0], "JToolTip");
+
+                    MetalToolTipUI toolTipObj = (MetalToolTipUI) MetalToolTipUI.createUI(tooltip);
+
+                    if (tooltip == null) {
+                        throw new RuntimeException("Metal Tooltip not been found");
+                    }
+                    checkAcclString(toolTipObj, tooltip);
+
+                } catch (Exception e) {
+                    throw new RuntimeException(e);
```

Comment:
Do you expect any other exception but those you throw yourself to fail the test?